### ISSUE TITLE
PrimaryGeneratedColumn - allow for override

### ISSUE THREAD

**Issue by twigs67** [typeorm/typeorm#7262]
Take the following scenario:

You want to migrate databases, and you need to keep primary column values the same. If you use PrimaryGeneratedColumn, you cannot override the column value as one is automatically generated for you. 

This is a huge problem. Perhaps there's a solution to this, but I haven't been able to find it.

**Comment by nebkat** [comment]
I believe if an ID value is provided (in `insert()`/`save()`/etc) it will be saved to the database and will not be overwritten by the generator, have you been experiencing otherwise?

**Comment by twigs67** [comment]
That's correct. When I add the id, it does not save when using save or insert. @nebkat

**Comment by Itsaadarsh** [comment]
Hey there @twigs67 well as far as I know there is a workaround for this but you'll lose the perquisite that `@PrimaryGeneratedColumn`  comes with, you can use `@PrimaryColumn` but every time you (save/insert) you need to run a function before that to increment the value it if you want.

**Comment by twigs67** [comment]
Am I missing something, this seems like a huge red flag and I believe the issue is with TypeOrm. Using .save() is not practical for all situation, e.g., bulk inserts. So anytime we use Query Builder, we need to run a function to update the increment sequence. That's bananas.

**Comment by bendadaniel** [comment]
+1

**Comment by chriszrc** [comment]
> That's correct. When I add the id, it does not save when using save or insert. @nebkat

It saves the entity, but it relies on the sequence and grabs the next id rather than using the provided id, even if that id doesn't currently exist in the table 😞

**Comment by buffcode** [comment]
Huge blocker when migrating data where IDs must be preserved, or when relying on third-party model definitions that cannot be modified.

When a value is given, the generator should take this value and not generate anything new.
